### PR TITLE
Some answers

### PR DESCRIPTION
**Pull request by Sopwith**
I may never fully understand binary, and I totally am gabbing out on persistence. But I remain hopeful I will someday be a useful member of computerized society.

### CODE REVIEW

File: hay-fraser/backup/compare-manifests.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import hashlib
+import csv
+
+def compare_manifests(manifest1 = r"C:\Users\Fraser\Desktop\Homework\a.csv", manifest2 = r"C:\Users\Fraser\Desktop\Homework\b.csv"):
```
Comment (gvwilson) [inline hay-fraser/backup/compare-manifests.py:4]: ouch - why the pathnames here?

File: hay-fraser/backup/compare-manifests.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    # Load the files
+    manifests = []
+    for m in [manifest1, manifest2]:
+        with open(m, 'r') as f:
+            reader = csv.reader(f)
+            next(reader) # Skip the header row
+            manifest = dict(row for row in reader)
+            manifests.append(manifest)
+    # Check: files with the same names but different hashes
+    for i in set(manifests[0].keys()) & set(manifests[1].keys()):
```
Comment (gvwilson) [inline hay-fraser/backup/compare-manifests.py:21]: nice

File: hay-fraser/backup/file_history.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import csv
+import glob
+
+# A bit of a mess but... Seemingly works... Basically the way I want it to...
+
+def file_history(file_name = "abc.txt", manifest_files = glob.glob(r"C:\Users\Fraser\Desktop\Homework\manifest*.csv")):
```
Comment (gvwilson) [inline hay-fraser/backup/file_history.py:6]: again, why the default value that's machine-specific?

**Review by gvwilson — APPROVED** [review]

File: hay-fraser/backup/compare-manifests.py (hunk 1 above)
Comment (Sopwith) [reply hay-fraser/backup/compare-manifests.py:4]: Ah, rough work. I should clean it out.

File: hay-fraser/backup/file_history.py (hunk 3 above)
Comment (Sopwith) [reply hay-fraser/backup/file_history.py:6]: What squalor!

File: hay-fraser/backup/compare-manifests.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    for i, h in manifests[0].items():
+        manifest_names1[h] = i
+    manifest_names2 = {}
+    for i, h in manifests[1].items():
+        manifest_names2[h] = i
+    for h in set(manifests[0].values()) & set(manifests[1].values()):
+        if manifest_names1[h] != manifest_names2[h]:
+            changes['renamed'].append((manifest_names1[h], manifest_names2[h]))
+    # Check: deleted files (from first to second) - exclude renames
+    for i in set(manifests[0].keys()) - set(manifests[1].keys()) - set(*changes['renamed']):
```
[the remark this answers is not in the document]
Comment (Sopwith) [reply hay-fraser/backup/compare-manifests.py:35]: I had never seen the "*" before until I was googling for this question. Don't know if I'll remember it next time I need it.

**Pull request closed without merging**